lib/python/rose/apps/ana_builtin/grepper.py

```diff
@@ -41,20 +45,34 @@ def run_command_and_check(self):
         self.command = self.command.format(*self.files)
         returncode, stdout, stderr = self.run_command(self.command)
         if returncode == 0:
-            self.parent.reporter(stdout, prefix="[INFO] ")
+            self.reporter(stdout, prefix="[INFO] ")
             self.passed = True
         else:
-            self.parent.reporter(stderr, prefix="[FAIL] ")
+            self.reporter(stderr, prefix="[FAIL] ")
+
+    def check_for_skip(self):
+        # If the users config options specified that the task should be
+        # ignored if all of its files were missing, set skipped attribute here
+        if self.skip_if_missing and len(self.files) > 0:
+            if all([not os.path.exists(fname) for fname in self.files]):
```

Comment:
He's right!  As can be seen:

```python
if any([time.sleep(5) is None for _ in range(1000)]):
    # Runs for 5,000 seconds
if any(time.sleep(5) is None for _ in range(1000)):
    # Exits after 5 seconds
```

I'll still update it to Matt's suggested arrangement as well